src/main/java/com/eppo/sdk/EppoClient.java

```diff
@@ -68,96 +51,153 @@ private Optional<EppoValue> getAssignmentValue(
         InputValidator.validateNotBlank(subjectKey, "Invalid argument: subjectKey cannot be blank");
         InputValidator.validateNotBlank(flagKey, "Invalid argument: flagKey cannot be blank");
 
+        VariationAssignmentResult assignmentResult = this.getAssignedVariation(flagKey, subjectKey, subjectAttributes);
+
+        if (assignmentResult == null) {
+            return Optional.empty();
+        }
+
+        Variation assignedVariation = assignmentResult.getVariation();
+        Optional<EppoValue> assignmentValue = Optional.of(assignedVariation.getTypedValue());
+
+        // Below is used for logging
+        String experimentKey = assignmentResult.getExperimentKey();
+        String allocationKey = assignmentResult.getAllocationKey();
+        String assignedVariationString = assignedVariation.getTypedValue().stringValue();
+
+        if (assignedVariation.getAlgorithmType() == AlgorithmType.OVERRIDE) {
```

Comment:
Not really familiar with Java. Are these get functions generated by lombok?

Reply:
They are. I'm not a fan of lombok automagic--even if it does save keystrokes--but it was already in this codebase, so I'm rolling with it. Happy to have us rip it out in a later PR 😅